src/App.js

```diff
@@ -1,25 +1,54 @@
-import logo from './logo.svg';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import React from 'react';
 import './App.css';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import axios from 'axios';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+export default class App extends React.Component{
+  constructor(props){
+    super(props);
+    this.state = {
+      searchQuery: '',
+      location: {},
+    }
+  }
+
+  getLocation = async (event) => {
+    event.preventDefault();
+    const url = `https://us1.locationiq.com/v1/search.php?key=${process.env.REACT_APP_CITY_KEY}&q=${this.state.searchQuery}&format=json`;
+    const response = await axios.get(url);
+    const location = response.data[0]
+    this.setState({
+      location,
+     
+  });
+  }
+  
 
-export default App;
+  render(){
+    return(
+      <>
+      <Form>
+        <Form.Label>
+          Enter a city name
+        </Form.Label>
+        <Form.Control onChange={(event) => this.setState({searchQuery: event.target.value})} type="text" placeholder="city name"/>
+        <Button onClick={this.getLocation} variant="secondary" type="Submit">
+          Explore!
+        </Button>
+        {this.state.location.place_id &&
+          <Form.Text> city: {this.state.location.display_name}</Form.Text>
+        }
+        {this.state.location.place_id &&
+          <Form.Text> lat: {this.state.location.lat}</Form.Text>
+        }
+        {this.state.location.place_id &&
+          <Form.Text> lon: {this.state.location.lon}</Form.Text>
+        }
+      </Form>
+      {/* <City location={this.state.location}/> */}
+      </>
+    )
+  }
+}
```

Comment:
Really nice work so far on this!

Feedback here:

- Remove setState call out of render (This was covered briefly in class, and can be used sometimes, but is not necessary in this case. Using it here is generally frowned upon as this can cause infinite loops and make component less maintainable. Instead create a method to handle event outside render and pass it to onChange.

- Consider cleaning the Form.Control up a bit. I see the line `this.state.location.place_id &&` is repeated several times. I bet you can figure out a way to do this only once 👍